### PR TITLE
String manipulation 2 - Yuliia P.

### PR DESCRIPTION
**Pull request by yuliia-p**
Screen Shot 2022-07-17 at 6 45 31 PM
Screen Shot 2022-07-17 at 6 46 26 PM
Screen Shot 2022-07-17 at 6 47 57 PM
Screen Shot 2022-07-17 at 6 49 04 PM

### CODE REVIEW

File: string-manipulation-2/src/capitalize-word.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/* exported capitalizeWord */
+// check if given word is 'javascript' word
+// before do all charters in lower or upper case
+// in if condition check if word is strikly equal to 'javascript' string
+// if yes, do first charater uppercase concatenate with next sectiom in lower case
+// and for charater at index 4 use touppercase again
+// if condition is false do next step - first letter to upper, rest lower
+function capitalizeWord(word) {
+  if (word.toUpperCase() === 'JAVASCRIPT') {
+    return word.charAt(0).toUpperCase() + word.slice(1, 4).toLowerCase() + word.charAt(4).toUpperCase() + word.slice(5).toLowerCase();
```
Comment (codingcodymiller) [inline string-manipulation-2/src/capitalize-word.js:10]: In special cases like this, you don't need to manually put this together. You can just `return "javaScript";`

File: string-manipulation-2/src/num-vowels.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/* exported numVowels */
+// using match method on string
+// retun of method is array of vowels
+// .length preperty give us a number of vowels
+
+function numVowels(string) {
+  if (!string.match(/[aeiou]/gi)) {
+    return 0;
+  } else {
+    var vowels = string.match(/[aeiou]/gi);
```
Comment (codingcodymiller) [inline string-manipulation-2/src/num-vowels.js:10]: You should do this without RegEx

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by yuliia-p** [comment]
Screen Shot 2022-07-18 at 10 15 31 PM
Screen Shot 2022-07-18 at 10 14 22 PM

File: string-manipulation-2/src/capitalize-word.js (hunk 1 above)
Comment (codingcodymiller) [inline string-manipulation-2/src/capitalize-word.js:10]: In special cases like this, you don't need to do the work of assembling the value if you already know what you need to return. You can just `return "JavaScript"` 🤓

File: string-manipulation-2/src/capitalize-words.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+// use join method with space
+// return new string
+
+function capitalizeWords(string) {
+  var newArr = [];
+  var words = string.split(' ');
+  for (var i = 0; i < words.length; i++) {
+    var capW = words[i].charAt(0).toUpperCase() + words[i].slice(1).toLowerCase();
+    newArr.push(capW);
+    var Str = newArr.join(' ');
```
Comment (codingcodymiller) [inline string-manipulation-2/src/capitalize-words.js:15]: You should do this after the loop has ended.

File: string-manipulation-2/src/num-vowels.js (hunk 2 above)
Comment (codingcodymiller) [reply string-manipulation-2/src/num-vowels.js:10]: This still needs to be addressed

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**